packages/gatsby-cli/src/index.js

```diff
@@ -17,9 +17,9 @@ const updateNotifier = require(`update-notifier`)
 // Check if update is available
 updateNotifier({ pkg }).notify()
 
-if (verDigit < 4) {
+if (verDigit < 6) {
   report.panic(
-    `Gatsby 1.0+ requires node.js v4 or higher (you have ${version}). \n` +
+    `Gatsby 1.0+ requires node.js v6 or higher (you have ${version}). \n` +
```

Comment:
Should change this to use the version of Gatsby too.

Reply:
Yhm, this is tricky, because `gatsby-cli` works with both gatsby 1 and gatsby 2 - I feel like this check should be done in gatsby core and not here ;) But I guess we set babel to target node 6 to build `gatsby-cli` now, so maybe that's just moot point

Reply:
We could also do a breaking change for the CLI and tell people on Gatsby V1 not to upgrade to the next major version. That way the CLI can support _only_ V2.

Reply:
@pieh and I discussed this, few things:

- We should improve this (in a separate PR)
- Gatsby cli (v1.1.58 at least) doesn't work with Node 4 anyways, so this change is reasonable
  - yurnalist is not compatible

I'm in favor of the breaking change, and also discussed perhaps documenting using npm scripts, e.g. `npm run develop`, `npm run build`, etc. which will use the project local version of Gatsby instead of global. That said, there are some concerns there, e.g. if a community starter doesn't include those scripts.

So more to do here, but relatively tangential to this PR!